docs/scripts/link-checker-puppeteer.ts

```diff
@@ -0,0 +1,46 @@
+import { sitePaths } from '../src/data/sitePaths';
+import {
+  checkLink,
+  crawlAllLinksFromAllPages,
+  runArrayPromiseInOrder,
+} from './util';
+
+if (![3, 4].includes(process.argv.length)) {
+  console.error('Expected 3 or 4 arguments!');
+  process.exit(1);
+}
+
+/**
+ * Divide the sitePaths array so that we can easily run a smaller portion if needed.
+ * e.g. to run the first 10 links. run `yarn docs node --require esbuild-register ./scripts/link-checker-puppeteer.ts 0 10`
+ */
+const start = process.argv[2];
+const end = process.argv[3];
+const testPaths = end ? sitePaths.slice(+start, +end) : sitePaths.slice(+start);
+try {
+  main();
+} catch (err) {
+  process.exit(1);
+}
+
+async function main() {
+  const allPagesPaths = await crawlAllLinksFromAllPages(testPaths);
+
+  await runArrayPromiseInOrder(
+    [...allPagesPaths],
```

Comment:
why is it important for these promises to run sequentially? a comment would be helpful.

Reply:
Done. 51cce098e692c9a208d61cf4ab251a24c348416c

Reply:
Why do we need to create a shallow copy of `allPagesPaths`?

Reply:
`allPagesPaths` is a `Map`

Reply:
Done a13e346c8393ac0654cde62bfca55d0e376d51df